src/replacer.js

```diff
@@ -83,20 +83,26 @@ export default class Replacer {
 						content = content.replaceAll(from, simpleTo ? to : (...args) => {
 							let resolvedArgs = resolveReplacementArgs(args);
 							to = this.to ?? this.insert ?? resolvedArgs.match;
+							let ret = resolvedArgs.match;
```

Comment:
Shouldn't `ret` be equal to `to` initially? Otherwise, if every check below fails, this code returns a different result than the previous code. Previously, we returned `to` that might not be equal to `resolvedArgs.match` in some cases. But here, we return `resolvedArgs.match` regardless.

Reply:
The idea was to avoid setting it to `to` unless we know that `to` is not a function. But you're right, this will break some cases. Let me push an attempt at a fix